src/components/views/rooms/Autocomplete.js

```diff
@@ -1,36 +1,48 @@
 import React from 'react';
 import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
 import classNames from 'classnames';
+import _ from 'lodash';
 
 import {getCompletions} from '../../../autocomplete/Autocompleter';
 
 export default class Autocomplete extends React.Component {
     constructor(props) {
         super(props);
+
+        this.onConfirm = this.onConfirm.bind(this);
+
         this.state = {
+            // list of completionResults, each containing completions
             completions: [],
 
+            // array of completions, so we can look up current selection by offset quickly
+            completionList: [],
+
             // how far down the completion list we are
-            selectionOffset: 0
+            selectionOffset: 0,
         };
     }
 
     componentWillReceiveProps(props, state) {
-        if(props.query == this.props.query) return;
+        if (props.query === this.props.query) {
+            return;
+        }
 
-        getCompletions(props.query, props.selection).map(completionResult => {
+        getCompletions(props.query, props.selection).forEach(completionResult => {
             try {
                 completionResult.completions.then(completions => {
                     let i = this.state.completions.findIndex(
                         completion => completion.provider === completionResult.provider
                     );
 
-                    i = i == -1 ? this.state.completions.length : i;
+                    i = i === -1 ? this.state.completions.length : i;
                     let newCompletions = Object.assign([], this.state.completions);
                     completionResult.completions = completions;
                     newCompletions[i] = completionResult;
+
                     this.setState({
-                        completions: newCompletions
+                        completions: newCompletions,
+                        completionList: _.flatMap(newCompletions, provider => provider.completions),
```

Comment:
OK - not opposed in principle, just want to make sure it's worth and extra 22kB on the bundle